src/pages/CreateAssignment/CreateAssignment.js

```diff
@@ -49,9 +49,46 @@ function CreateAssignment(props) {
     setLessonCreationDate(originalDate);
   }, [firebase]);
 
+  async function parseCardsFromLesson(lesson) {
+    let lessonCards = {};
+    let deploymentAccountIds = new Set();
+    for (const dueDate in lesson.dueDates) {
+      for (const deploymentAccount of lesson.dueDates[dueDate]) {
+        deploymentAccountIds.add(deploymentAccount);
+      }
+    }
+
+    let deploymentNameMap = {};
+    Array.from(deploymentAccountIds).forEach(id => {
+      firebase
+        .getDeploymentAccountInformation(id)
+        .then(deploymentAccount => {
+          deploymentNameMap[id] = deploymentAccount.username;
+        })
+        .catch(error => console.error("Error getting custom lesson: ", error));
+    });
+
+    // Wait for deploymentNameMap to finish resolving
+    while (Object.keys(deploymentNameMap).length == 0) {
+      await new Promise(r => setTimeout(r, 500));
+    }
```

Comment:
is this a bug?

Reply:
Yuppp, this is a react-specific bug where deploymentNameMap doesn't update quick enough, so we have to wait for it to update before I can use it

Reply:
can you add this to the refactor issue #56

Reply:
this won't work without the timeout delay since the object itself is undefined without time for it to update